### DISCUSSION TITLE
importing textgrids containing re-annotated transcriptions for LENA Lyon

### DISCUSSION THREAD

**Discussion by alecristia** [LAAC-LSCP/ChildProject#209] [Show and tell]
in this post I'm documenting how I imported textgrids for the LENA Lyon corpus that is part of EL1000, and originally hosted on homebank. 

# Getting the current version of the database

I started from the [README of EL1000](https://gin.g-node.org/EL1000/EL1000/src/main/README.md):

- I don't need to request access to the data
- or create an account, or give my username to the tech advisor, or install childproject
- or configuring ssh on gin 

I've done all of that before, so I start by adapting the instructions to install the datasets. 

I don't need to install the superdataset because I already have it, but it doesn't contain lyon (or many other datasets that are now available), since it may be a bit old. So I navigate to it and update it as follows:

```
cd Documents/git-data/EL1000/
source ~/ChildProjectVenv/bin/activate
datalad update
```
It turns out that it still doesn't contain the new datasets, so I read on and see that there are instructions for fetching updates:

```
datalad update --merge
datalad get .
```

Next, I navigate into lyon and set it up, as per the same instructions:

```
cd lyon
datalad run-procedure setup
```
(since these are homebank data, no need to setup confidential)

I run into a problem (#208) so I exited the superdataset and installed only lyon, following [these instructions](https://childproject.readthedocs.io/en/latest/reuse.html):

```
cd ../..
datalad install git@gin.g-node.org:/EL1000/lyon.git
cd lyon
ls
```

The contents & structure look ok. I read the warning in the instructions and checked, but there is no readme in lyon, so I think I'm ready to go.

# Preparation work

I've retrieved the annotations from oberon, and noticed:

- they are textgrids
- tiers are named based on a narrow speaker id (eg Brother, CHI, FEM, Mother, etc.) -- ie not just the 4 usual classes (CHI OCH FEM MAL)
- the file naming convention is: "child's first name (or part of it) " underscore "onset of the recording in local time in format HHhMM" -- eg Sazong_10h05.TextGrid
- this means I'll need to:
    - match up children's name to anonymized child ID
    - match up start time in local time to audio start time
- also, there is an error for a child who has data at 1am -- looking through the transcriptions, it seems that it's probably 13h05
- same child, there are some that say am and others pm
- other errors: 10h (rather than 10h00), 8h, 17h etc
- another child has 5h10 and 5h15 -- looks like these should be 17h etc

Given all of this, I decide to have one script that does the cleaning of naming errors & renaming so that we can have seek time referenced to the audio.  I'll be reusing an example where I did something similar in the past, [here](https://gin.g-node.org/EL1000/png2019/src/main/scripts/prep_eaf_import.sh).

For this, I need the raw .its for the files. So I find the command to get that by adapting one given within the [EL1000 README](https://gin.g-node.org/EL1000/EL1000/src/main/README.md)

```
 datalad get annotations/its/raw/. -J 4
```

I also notice that the names don't match either homebank nor broad_strokes. So since we've already done quite a bit of work matching kids up within lyon for a vtc re-analysis, I start from `/Users/acristia/Box/data_broad_strokes_lena/_rttm4reana/lyon_vtc/lyon_dict_matched.csv`. I still don't know how to match up the child's names to their homebank three-letter code (or child id from broadstrokes). Fortunately, MTLN has provided csv's that allow matching up the children to the wav/its from the broad strokes project: see `extra/humanval/tgfolder2wav.csv`.

- the textgrids were on `~/Desktop/LENA-LYON/*/*d`

Still to be done is to extract what the time clock appearing in the file name corresponds to within .its -- I'll just look up the names and times by hand (!) to compose a timecorr.txt table -- gave up! there are too many files (180).

**STOPPED HERE**

- todo: import vtc in box: `data_broad_strokes_lena/_rttm4reana/lyon_vtc/`
- todo: import cha -- a majority (but not all? weird...) are in oberon: `/scratch1/projects/lena_lyon/raw_LYON-master.zip` -- also, this looks like a git folder name, but I don't have it among my folders (it may belong to georgia and set to private)
- todo: import .textgrid -- these are day 1's of all the children, so partially the same data as cha, but where the boundaries and the text has been corrected. they are in oberon: `/scratch1/projects/lena_lyon/LENA_LYON_first_day/LENA-LYON`

**Comment by sarpu** [comment]
I think I got the vtc import, but I do have a question. What would be the proper function for lyon for figuring out the __filter__? For reference,  [this is what I am talking about for **bergelson**.](https://gin.g-node.org/EL1000/bergelson/src/main/scripts/import.py#L35-L41) And just to double check, I am using the **bergelson** import script as reference.

**Reply by sarpu** [reply]
Nvm those are just the individual `rttm` filenames, I figured it out (I think).

**Comment by alecristia** [comment]
if at any point, one gets something like:

> [ERROR  ] Unknown push target 'gin'. Known targets: 'origin'. [publish(/Users/acristia/Documents/git-data/lyon)] 
publish(error): . [Unknown push target 'gin'. Known targets: 'origin'.]

that means you should have done

`datalad push`

(`not datalad push --to gin`)
